README.md

```diff
@@ -145,3 +145,17 @@ use ApplicationInsights::Rack::TrackRequest, '<YOUR INSTRUMENTATION KEY GOES HER
 # For rails, suggest to set up this middleware in application.rb so that unhandled exceptions from controllers are also collected
 config.middleware.use 'ApplicationInsights::Rack::TrackRequest', '<YOUR INSTRUMENTATION KEY GOES HERE>', <buffer size>
 ```
+
+#### Retrieving a request's AppInsight ID  ####
+```ruby
+# from time to time you may need to access a request's AppInsight id from within your app
+application_insights_request_id = env['ApplicationInsights.request.id']
+
+# this can be used for a number of different purposes, including telemetry correlation
+uri = URI('http://api.example.com/search/?q=test')
+
+req = Net::HTTP::Get.new(uri)
+req['Request-Id'] = "|#{application_insights_request_id}." if application_insights_request_id

```

Comment:
I'm afraid correlation doesn't work here. Have you tested it to be working?
Basically, when you have multiple outgoing request, you want to increase the sequence, so it's like
request_id.1
request_id.2

Reply:
This was only meant as a superficial example more than anything but I will update it to show with a `"|#{application_insights_request_id}.1"`